caron.js

```diff
@@ -2,270 +2,253 @@
-    'payload = cjson.encode(payload)',
-    'payload = string.gsub(payload, \'"ARRAY_EMPTY"\', "[]")',
-    'payload = string.gsub(payload, \':{}\', ":null")',
-    'redis.call("SADD", "' + program.q_prefix + 'queues", jqueue)'
-  ].join('\n')
-}
+class Caron {
+  constructor (opts) {
+    this.status = {
+      active: 1,
+      processing: 0,
+      rseed: Date.now()
+    }
 
-scripts.bull = {
-  lua: [
-    'local pushCmd = "' + (program.q_lifo ? 'R' : 'L') + 'PUSH"',
-    'local jobId = redis.call("INCR", "bull:" .. jqueue .. ":id")',
-    'local jattempts = cmsg["$attempts"]',
-    'local jdelay = cmsg["$delay"]',
-    'local jobCustomId = cmsg["$jobId"]',
-    'if not(jobCustomId == nil or jobCustomId == "") then jobId = jobCustomId end',
-    'cmsg["$attempts"] = nil',
-    'cmsg["$delay"] = nil',
-    'cmsg["$jodId"] = nil',
-    'if (type(jattempts) ~= "number") or (jattempts <= 0) then jattempts = ' + program.def_attempts + ' end',
-    'if (type(jdelay) ~= "number") or (jdelay < 0) then jdelay = 0 end',
-    'local jopt0 = cmsg["$removeOnComplete"]',
-    'if (type(jopt0) ~= "boolean") then jopt0 = true end',
-    'local jopts = { removeOnComplete = jopt0 }',
-    'cmsg["$removeOnComplete"] = nil',
-    'local jobIdKey = "bull:" .. jqueue .. ":" .. jobId',
-    'if redis.call("EXISTS", jobIdKey) == 0 then',
-    '  redis.call("HMSET", jobIdKey, "data", cjson.encode(cmsg), "opts", cjson.encode(jopts), "progress", 0, "delay", jdelay, "timestamp", ARGV[2], "attempts", jattempts, "attemptsMade", 0, "stacktrace", "[]", "returnvalue", "null")',
-    '  if jdelay > 0 then',
-    '    local timestamp = (tonumber(ARGV[2]) + jdelay) * 0x1000 + bit.band(jobId, 0xfff)',
-    '    redis.call("ZADD", "bull:" .. jqueue .. ":delayed", timestamp, jobId)',
-    '    redis.call("PUBLISH", "bull:" .. jqueue .. ":delayed", (timestamp / 0x1000))',
-    '  else',
-    '    if redis.call("EXISTS", "bull:" .. jqueue .. ":meta-paused") ~= 1 then',
-    '      redis.call(pushCmd, "bull:" .. jqueue .. ":wait", jobId)',
-    '    else',
-    '      redis.call(pushCmd, "bull:" .. jqueue .. ":paused", jobId)',
-    '    end',
-    '    redis.call("PUBLISH", "bull:" .. jqueue .. ":waiting@null", jobId)',
-    '  end',
-    'end'
-  ].join('\n')
-}
+    this.debug = opts.debug
+    if (this.debug) console.log('started')
 
-scripts.sidekiq = {
-  lua: scripts.ruby_common_1 + '\n' + [
-    'redis.call("LPUSH", "' + program.q_prefix + 'queue:" .. jqueue, payload)'
-  ].join('\n')
-}
+    this.ptype = opts.type
+    this.freq = opts.freq
+    this.kill = opts.kill || true
 
-var script = [scripts.prefix, scripts[ptype].lua, scripts.suffix].join('\n')
+    this.redis = Redis.createClient(opts.redis)
+    this.registerHandlers()
 
-redis.defineCommand('qwork', {
-  lua: script,
-  numberOfKeys: 0
-})
+    const scripts = this.setupScripts(opts)
+    this.setupRedis(scripts)
 
-const elapsedTime = (start) => {
-  return process.hrtime(start)
-}
+    return this
```

Comment:
This is unnecessary

Reply:
yeah, made for some benchmark, was lazy to remove